### ISSUE TITLE
Index signature of object with type list of properties

### ISSUE THREAD

**Issue by MasGaNo** [Microsoft/TypeScript#12079]
**TypeScript Version:**  2.0.7

After set the `noImplicitAny` option to true in tsconfig.json, I get an "Index signature of object type implicitly has an 'any' type" even if I define the exact list of index key. 

**Code**

```ts
type HttpMethod = 'get'|'post'|'head'|'put'|'patch'|'delete'|'options';
interface IRouter {
    get: (path: string, callback: any) => {};
    post: (path: string, callback: any) => {};
    head: (path: string, callback: any) => {};
    put: (path: string, callback: any) => {};
    patch: (path: string, callback: any) => {};
    delete: (path: string, callback: any) => {};
    options: (path: string, callback: any) => {};
}

let router: IRouter;
let method: HttpMethod = 'post';

router[method]('foo', () => {});
```

**Expected behavior:**
As I define all properties that could be used dynamically, I expect that tsc can work with these values:
```
> tsc
>
```
**Actual behavior:** 
```
> tsc
Index signature of object type implicitly has an 'any' type
```

Thank you.

**Comment by ahejlsberg** [comment]
This works as you expect in the current nightly builds. It was fixed as part of #11929.

**Comment by MasGaNo** [comment]
Wow! Amazing! 

My bad, I don't use correct keyword to find this issue.

Great, I'm waiting for the new release, thank you :)
